appinventor/appengine/src/com/google/appinventor/client/editor/youngandroid/YaBlocksEditor.java

```diff
@@ -378,6 +378,14 @@ public static String getComponentInstanceTypeName(String formName, String instan
       return blocksEditor.myFormEditor.getComponentInstanceTypeName(instanceName);
   }
 
+  public static String getComponentInstancePropertyValue(String formName, String instanceName, String propertyName){
+    //use form name to get blocks editor
+    YaBlocksEditor blocksEditor = formToBlocksEditor.get(formName);
+    Map<String, MockComponent> componentMap = blocksEditor.myFormEditor.getComponents();
+    MockComponent mockComponent = componentMap.get(instanceName);
+    return mockComponent.getPropertyValue(propertyName);
```

Comment:
What if the component does not exists? Should we handle it instead of causing Null exceptions

Reply:
@ColinTree The only path to this code goes through the blocks drawer, which would need to be opened by clicking on the component instance in the source structure explorer. If the component doesn't exist, there would have been nothing to click on to open the drawer, and this code could not be triggered.

@ejyuen You might want to include a Javadoc that makes explicit the description I provided above and that calling this from another context may result in undefined behavior.